linux/rollup/ioctl-echo-loop/ioctl-echo-loop.c

```diff
@@ -102,7 +104,7 @@ static int finish_request(int fd, struct rollup_finish *finish, bool accept) {
     finish->accept_previous_request = accept;
     res = ioctl(fd, IOCTL_ROLLUP_FINISH, (unsigned long) finish);
     if (res != 0) {
-        fprintf(stderr, "IOCTL_ROLLUP_FINISH returned error %d\n", res);
+        fprintf(stderr, "IOCTL_ROLLUP_FINISH returned error %s [%d]\n", strerror(errno), errno);
```

Comment:
To be very picky, this doesn't work because strerror could modify errno. :/ So you should save it.

Reply:
Right, It may give a `strerror` errno code instead of what was expected.